ci/Jenkinsfiles/release.groovy

```diff
@@ -0,0 +1,215 @@
+            # to be removed when running with jenkins user
+            echo '{ "allow_root": true }' > /home/jenkins/.bowerrc
+          """
+        }
+      }
+    }
+
+    stage('Release') {
+      steps {
+        container('maven') {
+          script {
+            sh "git checkout -b release"
+            echo """
+            ----------------------------------------
+            Update parent version
+            ----------------------------------------
+            """
+            echo "MAVEN_OPTS=$MAVEN_OPTS"
+            sh "mvn ${MAVEN_ARGS} -V versions:update-parent -DgenerateBackupPoms=false"
+
+            def releaseVersion = readMavenPom().getParent().getVersion()
+            echo """
+            ----------------------------------------
+            Update version
+            ----------------------------------------
+            New version: ${releaseVersion}
+            """
+            sh """
+              mvn ${MAVEN_ARGS} versions:set -DnewVersion=${releaseVersion} -DgenerateBackupPoms=false
+              perl -i -pe 's|<nuxeo.jsf.version>.*?</nuxeo.jsf.version>|<nuxeo.jsf.version>${releaseVersion}</nuxeo.jsf.version>|' pom.xml
+
+              git commit -a -m "Release ${releaseVersion}"
+              git tag -a v${releaseVersion} -m "Release ${releaseVersion}"
+            """
+
+            echo """
+            ----------------------------------------
+            Build and Test
+            ----------------------------------------
+            """
+            sh "mvn ${MAVEN_ARGS} install"
+
+            if (env.DRY_RUN != "true") {
+              sh """
+                jx step git credentials
+                git config credential.helper store
+
+                git push origin v${releaseVersion}
+              """
+              currentBuild.description = "Release ${releaseVersion}"
+            }
+          }
+        }
+      }
+    }
+
+    stage('Deploy Maven artifacts') {
+      when {
+        not {
+          environment name: 'DRY_RUN', value: 'true'
+        }
+      }
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Deploy Maven artifacts
+          ----------------------------------------
+          """
+          sh """
+            mvn ${MAVEN_ARGS} -DskipTests deploy
+          """
+        }
+      }
+    }
+
+    stage('Upload Nuxeo Packages') {
+      when {
+        not {
+          environment name: 'DRY_RUN', value: 'true'
+        }
+      }
+      steps {
+        container('maven') {
+          echo """
+          ----------------------------------------
+          Upload Nuxeo Packages to ${CONNECT_PROD_URL}
+          ----------------------------------------
+          """
+          withCredentials([usernameColonPassword(credentialsId: 'connect-prod', variable: 'CONNECT_PASS')]) {
+            sh """
+              PACKAGES_TO_UPLOAD="packages/nuxeo-*-package/target/nuxeo-*-package-*.zip"
```

Comment:
Might not work for the `nuxeo-jsf-ui` package itself: `packages/nuxeo-marketplace-jsf-ui/target/nuxeo-marketplace-jsf-ui-11.2-SNAPSHOT.zip`.

Reply:
Good catch!